reactor-core/src/main/java/reactor/core/publisher/FluxIterable.java

```diff
@@ -158,11 +154,51 @@ static <T> void subscribe(CoreSubscriber<? super T> s, Spliterator<? extends T>
 		}
 
 		if (s instanceof ConditionalSubscriber) {
-			s.onSubscribe(new IterableSubscriptionConditional<>((ConditionalSubscriber<? super T>) s,
-					sp, knownToBeFinite, onClose));
+			IterableSubscriptionConditional<? extends T> isc =
+					new IterableSubscriptionConditional<>((ConditionalSubscriber<? super T>) s,
+							sp,
+							knownToBeFinite,
+							onClose);
+
+			boolean hasNext;
+			try {
+				hasNext = isc.hasNext();
+			}
+			catch (Throwable ex) {
+				Operators.error(s, ex);
+				isc.onCloseWithDropError();
```

Comment:
is it expected even before onSubscribe?

Reply:
look at `Operators.error(s, ex);` it looks like the following 

```java
s.onSubscribe(EmptySubscription.INSTANCE);
s.onError(e);
```

so it is after

Reply:
Got it, thanks.

Reply:
in that case, instantiated IterableSubscription is rather an existing thing that can close the resource instead of writing  

```java
             if (onClose != null) {
				try {
					onClose.run();
				}
				catch (Throwable t) {
					Operators.onErrorDropped(t, s.currentContext());
				}
			}
			```